src/main/java/com/mraof/minestuck/entity/item/ReturningProjectileEntity.java

```diff
@@ -0,0 +1,105 @@
+package com.mraof.minestuck.entity.item;
+
+import com.mraof.minestuck.client.renderer.entity.RendersAsItem;
+import com.mraof.minestuck.item.MSItems;
+import net.minecraft.entity.*;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.entity.projectile.ProjectileItemEntity;
+import net.minecraft.item.Item;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.IPacket;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.*;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.network.NetworkHooks;
+
```

Comment:
I am not entierly clear on how this entity is meant to behave. Perhaps you could write a javadoc on the code describing the projectiles intended behavior?

Reply:
If you want to, you could write one for the other entity class as well
But I think it's primarily valuable to have for this class

Reply:
I could make a description on the discord in showcase if you would like!